src/System Application/App/User Details/src/UserDetails.Table.al

```diff
@@ -0,0 +1,135 @@
+    fields
+    {
+        /// <summary>
+        /// An ID that uniquely identifies the user.
+        /// </summary>
+        field(1; "User Security ID"; Guid)
+        {
+            Access = Internal;
+        }
+        /// <summary>
+        /// User's name.
+        /// </summary>
+        field(2; "User Name"; Code[50])
+        {
+            CalcFormula = lookup(User."User Name" where("User Security ID" = field("User Security ID")));
+            Editable = false;
+            FieldClass = FlowField;
+            Access = Internal;
+        }
+        /// <summary>
+        /// User's full name.
+        /// </summary>
+        field(3; "Full Name"; Text[80])
+        {
+            CalcFormula = lookup(User."Full Name" where("User Security ID" = field("User Security ID")));
+            Editable = false;
+            FieldClass = FlowField;
+            Access = Internal;
+        }
+        /// <summary>
+        /// Specifies whether the user can access companies in the current environment.
+        /// </summary>
+        field(4; State; Option)
+        {
+            OptionCaption = 'Active,Inactive';
+            OptionMembers = Enabled,Disabled;
+            CalcFormula = lookup(User.State where("User Security ID" = field("User Security ID")));
+            Editable = false;
+            FieldClass = FlowField;
+            Access = Internal;
+        }
+        /// <summary>
+        /// The authentication email of the user.
+        /// </summary>
+        field(11; "Authentication Email"; Text[250])
+        {
+            CalcFormula = lookup(User."Authentication Email" where("User Security ID" = field("User Security ID")));
+            Editable = false;
+            FieldClass = FlowField;
+            Access = Internal;
+        }
+        /// <summary>
+        /// The contact email of the user.
+        /// </summary>
+        field(14; "Contact Email"; Text[250])
+        {
+            CalcFormula = lookup(User."Contact Email" where("User Security ID" = field("User Security ID")));
+            Editable = false;
+            FieldClass = FlowField;
+            Access = Internal;
+        }
+        /// <summary>
+        /// An ID that uniquely identifies the user for the purposes of sending telemetry.
+        /// </summary>
+        field(21; "Telemetry User ID"; Guid)
+        {
+            Caption = 'Telemetry ID';
+            CalcFormula = lookup("User Property"."Telemetry User ID" where("User Security ID" = field("User Security ID")));
+            Editable = false;
+            FieldClass = FlowField;
+            Access = Internal;
+        }
+        /// <summary>
+        /// An ID assigned to the user in Microsoft Entra.
+        /// </summary>
+        field(22; "Authentication Object ID"; Text[80])
+        {
+            Caption = 'Entra Object ID';
```

Comment:
Normally all the properties are sorted (check out e.g. _table 8889 "Sent Email"_).
If you're using AL Code Outline, then there's a code action to align this with other objects.

Reply:
Didn't know about this extension. Seems to have a lot of nice features - maybe we'll start using it internally.

Reply:
I think one of the best command is CleanUp on Uncommitted Files in Project.
You have to set the task in the settings.json (https://github.com/anzwdev/al-code-outline#settings) first, but then every developer can perform this task before committing and publishing.
![image](https://github.com/microsoft/BCApps/assets/56249171/0d99d73b-ead3-4f84-995c-1d8e835b1aab)